api/v1beta1/machinedeployment_types.go

```diff
@@ -25,6 +25,10 @@ const (
 	// MachineDeploymentTopologyFinalizer is the finalizer used by the topology MachineDeployment controller to
 	// clean up referenced template resources if necessary when a MachineDeployment is being deleted.
 	MachineDeploymentTopologyFinalizer = "machinedeployment.topology.cluster.x-k8s.io"
+
+	// MachineDeploymentFinalizer is the finalizer used by the MachineDeployment controller to
+	// cleanup the MachineDeployment descendant MachineSets when a MachineDeployment is being deleted.
+	MachineDeploymentFinalizer = "cluster.x-k8s.io/machinedeployment"
```

Comment:
adapted naming to what the latest state at this discussion + the linked slack was: 

https://github.com/kubernetes-sigs/cluster-api/pull/10791#discussion_r1740840138

(same for machineset)